// react-bulma-components/lib/components/section/section.story.js
import React from 'react';
import { Section, Container, Heading } from '../..';
export var Default = function Default(args) {
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Section, args, /*#__PURE__*/React.createElement(Container, null, /*#__PURE__*/React.createElement(Heading, null, "Section"), /*#__PURE__*/React.createElement(Heading, {
    subtitle: true
  }, "A simple container to divide your page into", ' ', /*#__PURE__*/React.createElement("strong", null, "sections"), ", like the one you are currently reading"))), /*#__PURE__*/React.createElement(Section, args, /*#__PURE__*/React.createElement(Container, null, /*#__PURE__*/React.createElement(Heading, null, "Section"), /*#__PURE__*/React.createElement(Heading, {
    subtitle: true
  }, "A simple container to divide your page into", ' ', /*#__PURE__*/React.createElement("strong", null, "sections"), ", like the one you are currently reading"))), /*#__PURE__*/React.createElement(Section, args, /*#__PURE__*/React.createElement(Container, null, /*#__PURE__*/React.createElement(Heading, null, "Section"), /*#__PURE__*/React.createElement(Heading, {
    subtitle: true
  }, "A simple container to divide your page into", ' ', /*#__PURE__*/React.createElement("strong", null, "sections"), ", like the one you are currently reading"))));
};
Default.argTypes = {
  size: {
    control: {
      type: 'select',
      options: ['medium', 'large']
    }
  }
};
