// react-bulma-components/lib/components/footer/footer.story.js
import React from 'react';
import { Footer, Container, Content, Hero } from '../..';
export var Default = function Default() {
  return /*#__PURE__*/React.createElement("div", {
    style: {
      margin: '-1rem'
    }
  }, /*#__PURE__*/React.createElement(Hero, {
    size: "fullheight"
  }, /*#__PURE__*/React.createElement(Hero.Header, {
    renderAs: "header"
  }), /*#__PURE__*/React.createElement(Hero.Body, null), /*#__PURE__*/React.createElement(Hero.Footer, null, /*#__PURE__*/React.createElement(Footer, null, /*#__PURE__*/React.createElement(Container, null, /*#__PURE__*/React.createElement(Content, {
    style: {
      textAlign: 'center'
    }
  }, /*#__PURE__*/React.createElement("p", null, /*#__PURE__*/React.createElement("strong", null, "Bulma"), " by", ' ', /*#__PURE__*/React.createElement("a", {
    href: "http://jgthms.com"
  }, "Jeremy Thomas"), ". The source code is licensed under", /*#__PURE__*/React.createElement("a", {
    href: "http://opensource.org/licenses/mit-license.php"
  }, ' ', "MIT"), ". The website content is licensed", ' ', /*#__PURE__*/React.createElement("a", {
    href: "http://creativecommons.org/licenses/by-nc-sa/4.0/"
  }, "CC BY NC SA 4.0"), ".")))))));
};
