// react-bulma-components/lib/components/dropdown/dropdown.js
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { if (typeof Symbol === "undefined" || !(Symbol.iterator in Object(arr))) return; var _arr = []; var _n = true; var _d = false; var _e = undefined; try { for (var _i = arr[Symbol.iterator](), _s; !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React, { useEffect, useRef, useState } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import CONSTANTS from '../../constants';
import DropdownItem from './components/item';
import DropdownDivider from './components/divider';
import Button from '../button';
import Element from '../element';
var colors = [null].concat(Object.values(CONSTANTS.COLORS));

var Dropdown = function Dropdown(_ref) {
  var className = _ref.className,
      children = _ref.children,
      value = _ref.value,
      color = _ref.color,
      align = _ref.align,
      right = _ref.right,
      up = _ref.up,
      hoverable = _ref.hoverable,
      label = _ref.label,
      onChange = _ref.onChange,
      closeOnSelect = _ref.closeOnSelect,
      icon = _ref.icon,
      domRef = _ref.domRef,
      props = _objectWithoutProperties(_ref, ["className", "children", "value", "color", "align", "right", "up", "hoverable", "label", "onChange", "closeOnSelect", "icon", "domRef"]);

  var ref = useRef(domRef);

  var _useState = useState(false),
      _useState2 = _slicedToArray(_useState, 2),
      isOpen = _useState2[0],
      setIsOpen = _useState2[1];

  var close = function close(evt) {
    // IDK yet how to test using the ref in enzime
    // istanbul ignore if
    if (hoverable || evt && ref && ref.current && ref.current.contains(evt.target)) {
      return;
    }

    if (ref.current) {
      setIsOpen(false);
    }
  };

  var onSelect = function onSelect(selectedValue) {
    return function () {
      if (onChange) {
        onChange(selectedValue);
      }

      if (closeOnSelect) {
        close();
      }
    };
  };

  useEffect(function () {
    window.addEventListener('click', close);
    return function () {
      window.removeEventListener('click', close);
    };
  }, []);
  var current = label;
  var childrenArray = React.Children.map(children, function (child, i) {
    if (child.type === DropdownItem && (i === 0 && !label || child.props.value === value)) {
      current = child.props.children;
    }

    return /*#__PURE__*/React.cloneElement(child, child.type === DropdownItem ? {
      active: child.props.value === value,
      onClick: onSelect(child.props.value)
    } : {});
  });
  return /*#__PURE__*/React.createElement(Element, _extends({}, props, {
    domRef: ref,
    className: classnames('dropdown', className, {
      'is-active': isOpen,
      'is-up': up,
      'is-right': right || align === 'right',
      'is-hoverable': hoverable
    })
  }), /*#__PURE__*/React.createElement("div", {
    className: "dropdown-trigger",
    role: "presentation",
    onClick: function onClick() {
      return setIsOpen(function (o) {
        return !o;
      });
    }
  }, /*#__PURE__*/React.createElement(Button, {
    color: color
  }, /*#__PURE__*/React.createElement("span", null, current), icon)), /*#__PURE__*/React.createElement("div", {
    className: "dropdown-menu",
    id: "dropdown-menu",
    role: "menu"
  }, /*#__PURE__*/React.createElement("div", {
    className: "dropdown-content"
  }, childrenArray)));
};

Dropdown.Item = DropdownItem;
Dropdown.Divider = DropdownDivider;
Dropdown.propTypes = _objectSpread(_objectSpread({}, Element.propTypes), {}, {
  /**
   * The value of the currently selected dropdown item. If this value match
   * with the value passed to a Dropdown.item it will be used as label if the label prop its empty
   */
  value: PropTypes.any,

  /**
   * Called when a dropdown item is selected.
   */
  onChange: PropTypes.func,

  /**
   * The color of the dropdown button.
   */
  color: PropTypes.oneOf(colors),

  /**
   * Whether the dropdown should align to the right side.
   */
  right: PropTypes.bool,

  /**
   * Whether the dropdown menu should appear above the dropdown button
   * instead of below.
   */
  up: PropTypes.bool,

  /**
   * Whether the dropdown menu can be activated when the cursor
   * hovers above the button without clicking.
   */
  hoverable: PropTypes.bool,

  /**
   * A string, or a react component that displays the label of the dropdown
   * button. if not set it will be use the selected Dropdown.Item or the first one
   * if there is no selected
   */
  label: PropTypes.node,

  /**
   * Whether the dropdown menu should be closed when a dropdown item is selected.
   */
  closeOnSelect: PropTypes.bool,

  /**
   * A react component that draws the icon of the dropdown button.
   * Usually it is an arrow (or a chevron) pointing downwards (or upwards).
   */
  icon: PropTypes.node
  /**
   * Additional CSS classes to pass to `<Dropdown />`.
   * They will sit alongside pre-applied bulma classes.
   */

});
Dropdown.defaultProps = {
  closeOnSelect: true
};
export default Dropdown;
