// react-bulma-components/lib/components/hero/hero.story.js
import React from 'react';
import { Hero, Heading, Container, Notification } from '../..';
import CONSTANTS from '../../constants';
export var Default = function Default(args) {
  return /*#__PURE__*/React.createElement("div", {
    style: {
      margin: '-1rem'
    }
  }, /*#__PURE__*/React.createElement(Hero, args, /*#__PURE__*/React.createElement(Hero.Header, {
    renderAs: "header"
  }, /*#__PURE__*/React.createElement(Notification, {
    color: "dark"
  }, "Header")), /*#__PURE__*/React.createElement(Hero.Body, null, /*#__PURE__*/React.createElement(Container, null, /*#__PURE__*/React.createElement(Heading, null, "Hero title Info"), /*#__PURE__*/React.createElement(Heading, {
    subtitle: true,
    size: 3
  }, "Subtitle"))), /*#__PURE__*/React.createElement(Hero.Footer, null, /*#__PURE__*/React.createElement(Notification, {
    color: "dark"
  }, "Footer"))));
};
Default.argTypes = {
  color: {
    control: {
      type: 'select',
      options: ['default'].concat(Object.values(CONSTANTS.COLORS))
    }
  },
  size: {
    control: {
      type: 'select',
      options: ['default', 'small', 'medium', 'large', 'fullheight']
    }
  }
};
