// react-bulma-components/lib/components/panel/panel.story.js
import React from 'react';
import { Panel, Form, Button } from '../..';
import CONSTANTS from '../../constants';
export var Default = function Default(args) {
  return /*#__PURE__*/React.createElement(Panel, args, /*#__PURE__*/React.createElement(Panel.Header, null, "Repositories"), /*#__PURE__*/React.createElement(Panel.Block, null, /*#__PURE__*/React.createElement(Form.Control, null, /*#__PURE__*/React.createElement(Form.Input, {
    size: "small",
    type: "text",
    placeholder: "search"
  }))), /*#__PURE__*/React.createElement(Panel.Tabs, {
    className: "panel-tabs"
  }, /*#__PURE__*/React.createElement(Panel.Tabs.Tab, {
    active: true
  }, "all"), /*#__PURE__*/React.createElement(Panel.Tabs.Tab, null, "public"), /*#__PURE__*/React.createElement(Panel.Tabs.Tab, null, "private"), /*#__PURE__*/React.createElement(Panel.Tabs.Tab, null, "sources"), /*#__PURE__*/React.createElement(Panel.Tabs.Tab, null, "forks")), /*#__PURE__*/React.createElement(Panel.Block, {
    renderAs: "a",
    active: true
  }, /*#__PURE__*/React.createElement(Panel.Icon, null, /*#__PURE__*/React.createElement("i", {
    className: "fas fa-angle-down"
  })), "bulma"), /*#__PURE__*/React.createElement(Panel.Block, null, /*#__PURE__*/React.createElement(Panel.Icon, null, /*#__PURE__*/React.createElement("i", {
    className: "fas fa-angle-down"
  })), "react-bulma-components"), /*#__PURE__*/React.createElement(Panel.Block, null, /*#__PURE__*/React.createElement(Panel.Icon, null, /*#__PURE__*/React.createElement("i", {
    className: "fas fa-angle-down"
  })), "minireset.css"), /*#__PURE__*/React.createElement(Panel.Block, null, /*#__PURE__*/React.createElement(Panel.Icon, null, /*#__PURE__*/React.createElement("i", {
    className: "fas fa-angle-down"
  })), "jgthms.github.io"), /*#__PURE__*/React.createElement(Panel.Block, {
    renderAs: "label"
  }, /*#__PURE__*/React.createElement(Form.Checkbox, null), "remember me"), /*#__PURE__*/React.createElement(Panel.Block, null, /*#__PURE__*/React.createElement(Button, {
    fullwidth: true,
    color: args.color,
    outlined: true
  }, "reset all filters")));
};
Default.argTypes = {
  color: {
    control: {
      type: 'select',
      options: ['default'].concat(Object.values(CONSTANTS.COLORS))
    }
  }
};
