// react-bulma-components/lib/components/box/box.story.js
import React from 'react';
import { Box, Form, Button } from '../..';
export var Default = function Default() {
  return /*#__PURE__*/React.createElement(Box, {
    style: {
      width: 400,
      margin: 'auto'
    }
  }, /*#__PURE__*/React.createElement("form", null, /*#__PURE__*/React.createElement(Form.Field, null, /*#__PURE__*/React.createElement(Form.Label, null, "Email"), /*#__PURE__*/React.createElement(Form.Control, null, /*#__PURE__*/React.createElement(Form.Input, {
    type: "email",
    placeholder: "my.amazing.email@react-bulma.dev"
  }))), /*#__PURE__*/React.createElement(Form.Field, null, /*#__PURE__*/React.createElement(Form.Label, null, "Password"), /*#__PURE__*/React.createElement(Form.Control, null, /*#__PURE__*/React.createElement(Form.Input, {
    type: "password",
    placeholder: "*************"
  }))), /*#__PURE__*/React.createElement(Button.Group, {
    align: "right"
  }, /*#__PURE__*/React.createElement(Button, {
    color: "primary"
  }, "Sign in"))));
};
