// react-bulma-components/lib/components/heading/heading.story.js
/* eslint-disable react/prop-types */
import React from 'react';
import Heading from './heading';
export var Default = function Default(_ref) {
  var titleSize = _ref.titleSize,
      titleWeight = _ref.titleWeight,
      spaced = _ref.spaced;
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Heading, {
    spaced: spaced,
    size: titleSize,
    weight: titleWeight
  }, "Title"), /*#__PURE__*/React.createElement(Heading, {
    subtitle: true
  }, "Subtitle"));
};
Default.argTypes = {
  titleSize: {
    name: 'size',
    table: {
      category: 'Title'
    },
    control: {
      type: 'select',
      options: ['default', 1, 2, 3, 4, 5, 6]
    }
  },
  titleWeight: {
    name: 'weight',
    table: {
      category: 'Title'
    },
    control: {
      type: 'select',
      options: ['default', 'light', 'normal', 'semibold', 'bold']
    }
  },
  spaced: {
    table: {
      category: 'Title'
    },
    control: {
      type: 'boolean'
    }
  }
};
