// react-bulma-components/lib/components/card/card.story.js
import React from 'react';
import { Card, Media, Image, Content, Heading } from '../..';
export var Default = function Default() {
  return /*#__PURE__*/React.createElement(Card, {
    style: {
      width: 300,
      margin: 'auto'
    }
  }, /*#__PURE__*/React.createElement(Card.Image, {
    size: "4by3",
    src: "http://bulma.io/images/placeholders/1280x960.png"
  }), /*#__PURE__*/React.createElement(Card.Content, null, /*#__PURE__*/React.createElement(Media, null, /*#__PURE__*/React.createElement(Media.Item, {
    renderAs: "figure",
    align: "left"
  }, /*#__PURE__*/React.createElement(Image, {
    size: 64,
    alt: "64x64",
    src: "http://bulma.io/images/placeholders/128x128.png"
  })), /*#__PURE__*/React.createElement(Media.Item, null, /*#__PURE__*/React.createElement(Heading, {
    size: 4
  }, "John Smith"), /*#__PURE__*/React.createElement(Heading, {
    subtitle: true,
    size: 6
  }, "@johnsmith"))), /*#__PURE__*/React.createElement(Content, null, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus nec iaculis mauris. ", /*#__PURE__*/React.createElement("a", null, "@bulmaio"), ".", /*#__PURE__*/React.createElement("a", {
    href: "#1"
  }, "#css"), ' ', /*#__PURE__*/React.createElement("a", {
    href: "#2"
  }, "#responsive"), /*#__PURE__*/React.createElement("br", null), /*#__PURE__*/React.createElement("time", {
    dateTime: "2016-1-1"
  }, "11:09 PM - 1 Jan 2016"))));
};
export var FooterActions = function FooterActions() {
  return /*#__PURE__*/React.createElement(Card, null, /*#__PURE__*/React.createElement(Card.Header, null, /*#__PURE__*/React.createElement(Card.Header.Title, null, "Title")), /*#__PURE__*/React.createElement(Card.Content, null, /*#__PURE__*/React.createElement(Media, null, /*#__PURE__*/React.createElement(Media.Item, {
    renderAs: "figure",
    align: "left"
  }, /*#__PURE__*/React.createElement(Image, {
    size: 64,
    alt: "64x64",
    src: "http://bulma.io/images/placeholders/128x128.png"
  })), /*#__PURE__*/React.createElement(Media.Item, null, /*#__PURE__*/React.createElement(Heading, {
    size: 4
  }, "John Smith"), /*#__PURE__*/React.createElement(Heading, {
    subtitle: true,
    size: 6
  }, "@johnsmith"))), /*#__PURE__*/React.createElement(Content, null, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus nec iaculis mauris. ", /*#__PURE__*/React.createElement("a", null, "@bulmaio"), ".", /*#__PURE__*/React.createElement("a", {
    href: "#1"
  }, "#css"), ' ', /*#__PURE__*/React.createElement("a", {
    href: "#2"
  }, "#responsive"), /*#__PURE__*/React.createElement("br", null), /*#__PURE__*/React.createElement("time", {
    dateTime: "2016-1-1"
  }, "11:09 PM - 1 Jan 2016"))), /*#__PURE__*/React.createElement(Card.Footer, null, /*#__PURE__*/React.createElement(Card.Footer.Item, {
    renderAs: "a",
    href: "#Yes"
  }, "Yes"), /*#__PURE__*/React.createElement(Card.Footer.Item, {
    renderAs: "a",
    href: "#No"
  }, "No"), /*#__PURE__*/React.createElement(Card.Footer.Item, {
    renderAs: "a",
    href: "#Maybe"
  }, "Maybe")));
};
FooterActions.storyName = 'With footer action';
