// react-bulma-components/lib/components/tile/tile.story.js
import React from 'react';
import { Tile, Heading, Image, Section, Box, Notification } from '../..';
export var Default = function Default() {
  return /*#__PURE__*/React.createElement(Section, null, /*#__PURE__*/React.createElement(Box, null, /*#__PURE__*/React.createElement(Tile, {
    kind: "ancestor"
  }, /*#__PURE__*/React.createElement(Tile, {
    size: 8,
    vertical: true
  }, /*#__PURE__*/React.createElement(Tile, null, /*#__PURE__*/React.createElement(Tile, {
    kind: "parent",
    vertical: true
  }, /*#__PURE__*/React.createElement(Tile, {
    kind: "child",
    renderAs: Notification,
    color: "primary"
  }, /*#__PURE__*/React.createElement(Heading, null, "Vertical..."), /*#__PURE__*/React.createElement(Heading, {
    subtitle: true
  }, "Top tile")), /*#__PURE__*/React.createElement(Tile, {
    kind: "child",
    renderAs: Notification,
    color: "warning"
  }, /*#__PURE__*/React.createElement(Heading, null, "Tiles..."), /*#__PURE__*/React.createElement(Heading, {
    subtitle: true
  }, "Bottom Tile..."))), /*#__PURE__*/React.createElement(Tile, {
    kind: "parent"
  }, /*#__PURE__*/React.createElement(Tile, {
    kind: "child",
    renderAs: Notification,
    color: "info"
  }, /*#__PURE__*/React.createElement(Heading, null, "Middle Tile..."), /*#__PURE__*/React.createElement(Heading, {
    subtitle: true
  }, "With image Tile..."), /*#__PURE__*/React.createElement(Image, {
    size: "4by3",
    src: "http://bulma.io/images/placeholders/640x480.png"
  })))), /*#__PURE__*/React.createElement(Tile, {
    kind: "parent"
  }, /*#__PURE__*/React.createElement(Tile, {
    kind: "child",
    renderAs: Notification,
    color: "danger"
  }, /*#__PURE__*/React.createElement(Heading, null, "Wide tile"), /*#__PURE__*/React.createElement(Heading, {
    subtitle: true
  }, "Aligned with the right tile"), /*#__PURE__*/React.createElement("div", {
    className: "content"
  })))), /*#__PURE__*/React.createElement(Tile, {
    kind: "parent"
  }, /*#__PURE__*/React.createElement(Tile, {
    kind: "child",
    renderAs: Notification,
    color: "success"
  }, /*#__PURE__*/React.createElement("div", {
    className: "content"
  }, /*#__PURE__*/React.createElement(Heading, null, "Tall tile"), /*#__PURE__*/React.createElement(Heading, {
    subtitle: true
  }, "With even more content"), /*#__PURE__*/React.createElement("div", {
    className: "content"
  })))))));
};
